### PR TITLE
Assign different storage config for indexes

### PR DESCRIPTION
**Pull request by xige-16**
/kind improvement
Signed-off-by: xige-16 <xi.ge@zilliz.com>

### CODE REVIEW

File: internal/querynode/load_index_info.go [hunk 1]
```diff
@@ -149,6 +149,34 @@ func (li *LoadIndexInfo) appendIndexData(bytesIndex [][]byte, indexKeys []string
...
+		access_key_id:    cAccessKey,
+		access_key_value: cAccessValue,
+		remote_root_path: cRootPath,
+		storage_type:     cStorageType,
+		iam_endpoint:     cIamEndPoint,
+		useSSL:           C.bool(Params.MinioCfg.UseSSL),
+		useIAM:           C.bool(Params.MinioCfg.UseIAM),
+	}
+
+	status = C.AppendIndex(li.cLoadIndexInfo, cBinarySet, storageConfig)
```
Comment (czs007) [inline internal/querynode/load_index_info.go:180]: suggest move storageConfig to cLoadIndexInfo

Comment (jennyli-z) [reply internal/querynode/load_index_info.go:180]: why?

File: internal/core/src/storage/DiskFileManagerImpl.h [hunk 2]
```diff
@@ -99,6 +107,9 @@ class DiskFileManagerImpl : public FileManagerImpl {
 
     // remote file path
     std::map<std::string, int64_t> remote_paths_to_size_;
+
+    MinioChunkManagerPtr rcm_;
```
Comment (czs007) [inline internal/core/src/storage/DiskFileManagerImpl.h:111]: change type to RemoteChunkManagerPtr?

File: internal/querynode/load_index_info.go (hunk 1 above)
Comment (czs007) [reply internal/querynode/load_index_info.go:180]: There is no need to expose one more parameter, you can directly specify storageConfig when constructing LoadIndexInfo.
The underlying code just takes the storageConfig from LoadIndexInfo.

File: internal/core/src/storage/DiskFileManagerImpl.h (hunk 2 above)
Comment (czs007) [reply internal/core/src/storage/DiskFileManagerImpl.h:111]: Use polymorphism

File: internal/core/src/storage/DiskFileManagerImpl.cpp [hunk 3]
```diff
@@ -200,12 +201,12 @@ DiskFileManagerImpl::RemoveFile(const std::string& file) noexcept {
     std::string remoteFile = "";
     bool remoteExist = false;
     FILEMANAGER_TRY
-    remoteExist = remote_chunk_manager.Exist(remoteFile);
+    remoteExist = rcm_->Exist(remoteFile);
     FILEMANAGER_CATCH
     FILEMANAGER_END
     if (!remoteExist) {
```
Comment (czs007) [inline internal/core/src/storage/DiskFileManagerImpl.cpp:207]: is this right ?

File: internal/core/src/storage/MinioChunkManager.cpp [hunk 4]
```diff
@@ -94,13 +89,19 @@ MinioChunkManager::MinioChunkManager(const std::string& endpoint,
...
-            Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never, false);
+            Aws::Auth::AWSCredentials(ConvertToAwsString(storage_config.access_key_id),
+                                      ConvertToAwsString(storage_config.access_key_value)),
+            config, Aws::Client::AWSAuthV4Signer::PayloadSigningPolicy::Never, false);
     }
 
-    LOG_SEGCORE_INFO_C << "init MinioChunkManager with parameter[endpoint: '" << endpoint << "', access_key:'"
-                       << access_key << "', access_value:'" << access_value << "', default_bucket_name:'" << bucket_name
-                       << "', use_secure:'" << std::boolalpha << secure << "']";
+    if (!BucketExists(storage_config.bucket_name)) {
```
Comment (czs007) [inline internal/core/src/storage/MinioChunkManager.cpp:97]: Is this thread safe?

**Pull request merged**